src/util/GeoSparqlHelpers.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright 2022, University of Freiburg,
+// Author: Hannah Bast <bast@cs.uni-freiburg.de>
+// Chair of Algorithms and Data Structures
+
+#include "./GeoSparqlHelpers.h"
+
+#include <absl/strings/charconv.h>
+#include <ctre/ctre.h>
+
+#include <cmath>
+#include <limits>
+#include <numbers>
+#include <optional>
+#include <string>
+#include <string_view>
+
+#include "./Exception.h"
+
+namespace ad_utility {
+
+namespace detail {
+
+static constexpr auto wktPointRegex = ctll::fixed_string(
+    "^\"[Pp][Oo][Ii][Nn][Tt] *\\( *"
+    "(-?[0-9]+|-?[0-9]+\\.[0-9]+) +(-?[0-9+]|-?[0-9]+\\.[0-9]+)"
+    " *\\)\"(\\^\\^|$)");
+
+static constexpr double invalidCoordinate =
+    std::numeric_limits<double>::quiet_NaN();
+
+// Parse a single WKT point and returns a pair of latitude and longitude (note
+// that in the WKT point the order is reverse). If the given string does not
+// parse as a WKT point, a pair of `invalidCoordinate` is returned.
+std::pair<double, double> parseWktPoint(const std::string& wktPoint) {
+  double lng = invalidCoordinate, lat = invalidCoordinate;
+  if (auto match = ctre::search<wktPointRegex>(wktPoint)) {
+    std::string_view lng_sv = match.get<1>();
+    std::string_view lat_sv = match.get<2>();
+    absl::from_chars(lng_sv.begin(), lng_sv.end() + lng_sv.size(), lng);
```

Comment:
There is a reason I did it this way: http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2020/p2007r0.html

However, I made a stupid mistake, I put end() + size() instead of begin() + size(), which is fixed now.

Please let me know if you think that in 2022, the problem explained on the website above is no longer a problem.